src/pytorch_lightning/loggers/wandb.py

```diff
@@ -288,17 +289,16 @@ def __init__(
         self._log_model = log_model
         self._prefix = prefix
         self._experiment = experiment
-        self._logged_model_time = {}
-        self._checkpoint_callback = None
+        self._logged_model_time: Dict[str, float] = {}
+        self._checkpoint_callback: Optional["ReferenceType[Checkpoint]"] = None
         # set wandb init arguments
-        anonymous_lut = {True: "allow", False: None}
-        self._wandb_init = dict(
+        self._wandb_init: Dict[str, Any] = dict(
```

Comment:
> Changing- `"notes"` to `"name"` like in the other logger tests fixes the issue. However I'm confused at how this test was even passing in the first place..

@gautierdag don't change the test. Precisely it is `"notes"` to catch regressions.

With the new version of jsonargparse there the source code is analyzed using ASTs to figure out which parameters are accepted. This supports a limited amount of code patterns, see [AST resolver](https://jsonargparse.readthedocs.io/en/stable/#ast-resolver). From what I see, adding a type hint in
`self._wandb_init: Dict[str, Any] = dict` makes it a different pattern. If adding this type hint is truly necessary, then this needs to wait until the support for it is added to jsonargparse.

Reply:
Thanks for the explanation! 
It's not truly necessary, I'll add a type ignore to make `mypy` happy and that should fix the CLI test as well.

Reply:
The fix is easy so just did it, https://github.com/omni-us/jsonargparse/commit/9eea0ad9ca65863e7ddb9081e0d1d43a2e5ce53a and released it as jsonargparse v2.10.2. If you like, you can keep the type hint, but then you need to update the minimum jsonargparse version in `requirements/pytorch/extra.txt` and in `src/pytorch_lightning/utilities/cli.py`.